src/issuer/SsiCredentialIssuer.Service/BusinessLogic/IssuerBusinessLogic.cs

```diff
@@ -85,10 +85,22 @@ public IssuerBusinessLogic(
     }
 
     /// <inheritdoc />
-    public IAsyncEnumerable<UseCaseParticipationData> GetUseCaseParticipationAsync() =>
-        _repositories
+    public IAsyncEnumerable<UseCaseParticipationData> GetUseCaseParticipationAsync(string? status)
+    {
+        StatusType? statusTypeResult = null;
+        if (!string.IsNullOrEmpty(status))
+        {
+            if (!(Enum.TryParse<StatusType>(status, ignoreCase: true, out var statusType) || !Enum.IsDefined(typeof(StatusType), statusType)))
+            {
+                throw new ArgumentException($"Status value {status} is not valid; please use Active, Expired or All");
+            }
+            statusTypeResult = statusType;
+        }
```

Comment:
You can directly use StatusType, than the conversion isn't needed

```suggestion
    public IAsyncEnumerable<UseCaseParticipationData> GetUseCaseParticipationAsync(StatusType? status)
    {
        
```

Reply:
Here is the same. When I use the type object I cannot have a customized message. At least I could not find a way to do that. I would be grateful to have a support here